src/test/scala/io/citrine/lolo/bags/BaggerTest.scala

```diff
@@ -360,17 +360,29 @@ class BaggerTest {
   def testShapley(): Unit = {
     rng.setSeed(24795L)
     val nCols = 5
-    val trainingData = TestUtils.generateTrainingData(1024, nCols, noise = 0.0, function = Friedman.friedmanSilverman, seed = rng.nextLong())
-    val DTLearner = RegressionTreeLearner(numFeatures = 2, rng = rng)
+    val trainingData = TestUtils.generateTrainingData(64, nCols, noise = 0.0, function = Friedman.friedmanSilverman, seed = rng.nextLong())
+    val DTLearner = RegressionTreeLearner(numFeatures = nCols, rng = rng)
     val model = Bagger(DTLearner, randBasis = TestUtils.getBreezeRandBasis(rng.nextLong()))
       .train(trainingData)
       .getModel()
+    val trees = model.getModels()
     trainingData.foreach { case (x, _) =>
       val shapley = model.shapley(x).get
+
+      // Do a quick sanity check on the output format.
       assert(shapley.length == nCols)
       assert(shapley.forall {
         _.length == 1
       })
+
+      // Compute the mean shap value over trees and ensure the bagged model gives the same result.
+      val treeMean = trees.map { t =>
+        t.shapley(x).get.map { _(0) / trees.length }
```

Comment:
I think you can use `flatten` here

Reply:
Flatten doesn't work in retrieving the doubles because of DenseVector.